### ISSUE TITLE
Async recursive function produces StackOverflowException where as .Net doesn't.

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#1398]
### Expected

run indefinitely.

### Actual

Uncaught RangeError: maximum call stack exceeded

### Steps To Reproduce
[Live Bridge](http://live.bridge.net/#b654e8998b5dd9f592da66d504428f46)
[Dot Net Fiddle](https://dotnetfiddle.net/4Yh7bc)
```csharp
/// <summary>
/// Creates a simple class
/// </summary>
public class App
{
    // Use the [Ready] attribute to automatically
    // run this method when the page is ready.
    [Ready]
    public static async void Main()
    {
        Main();
    }
}
```

**Comment by LeonidVeriga** [comment]
@michaelcheers,

The [Dot Net Fiddle](https://dotnetfiddle.net/4Yh7bc) sample gives the same recursive problem:
`[DotNetFiddle.Infrastructure.LimitExceededException: Execution time limit was exceeded]`

**Comment by michaelcheers** [comment]
Closes #1398.
